types/src/reconfiguration_event_subscription.rs

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+/// Trait that must be implemented by a subscription to reconfiguration events emitted
+/// by state sync
+
+pub trait ReconfigurationEventSubscription: Send + Sync {
```

Comment:
I don't like long names, how about just `EventSubscription`? it sounds more general

Reply:
do we have plans for having more types of subscriptions? candidates for names right now are `EventSubscription` and `ReconfigSubscriber`. If this is the only event type I can choose the former since it's more generic

Reply:
`EventSubscriber` : P

Reply:
ok, going with `EventSubscription` and we can expand later to potentially support more types of events